src/test/ui/typeck/issue-91328.fixed

```diff
@@ -0,0 +1,27 @@
+// Regression test for issue #91328.
```

Comment:
Might be nice to see examples where the type being matched is *not* a `Result<Vec<_>>` or  `Option<Vec<_>>`. I'm thinking straight `Vec<_>` or some other type that doesn't deref to slice. (Though, not wrapping in a `Result`/`Option` might be covered elsewhere)

Reply:
Clarification: this is the only thing I'd like to see before this PR is merged, but doesn't *need* to block this PR.

Reply:
I've added a test case with a match on a plain `Vec<_>` now (sorry for taking so long to get around to this...).